commit eecdcabac84dd93abf026fbfdb6f1c850c98fa5b
Author: Fabio M. Di Nitto <fdinitto@redhat.com>
Date:   Fri Aug 5 09:56:26 2011 +0200

    cman: fix printing of error if clustername is too long
    
    Patch 03e9af7db105 did mask an error and was not propagating the
    error string in the cman pipe, making it impossible to see the error
    at startup when it is more important, since clustername cannot be changed
    at runtime.
    
    Resolves: rhbz#728230
    
    Signed-off-by: Christine Caulfield <ccaulfie@redhat.com>
    Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

diff --git a/cman/daemon/cman-preconfig.c b/cman/daemon/cman-preconfig.c
index 0abc4cd..d91dce2 100644
--- a/cman/daemon/cman-preconfig.c
+++ b/cman/daemon/cman-preconfig.c
@@ -1290,13 +1290,15 @@ static int get_cman_globals(struct objdb_iface_ver0 *objdb)
 	char *use_hash;
 
 	objdb_get_string(objdb, cluster_parent_handle, "name", &cluster_name);
-	if (!cluster_name || strlen(cluster_name) > 15) {
-		    sprintf(error_reason, "%s\n", "Invalid cluster name. It must be 15 characters or fewer\n");
+	if (!cluster_name) {
+		sprintf(error_reason, "Unable to determine cluster name.\n");
+		write_cman_pipe("Unable to determine cluster name.\n");
 		return -1;
 	}
 
-	if (!cluster_name) {
-		sprintf(error_reason, "Unable to determine cluster name.\n");
+	if (strlen(cluster_name) > 15) {
+		sprintf(error_reason, "%s\n", "Invalid cluster name. It must be 15 characters or fewer\n");
+		write_cman_pipe("Invalid cluster name. It must be 15 characters or fewer\n");
 		return -1;
 	}
 
